modules/tensor_mechanics/include/materials/NeoHookeanStressUpdate.h

```diff
@@ -0,0 +1,34 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ComputeLagrangianStressPK2.h"
+
+/// Compressible Neo-Hookean hyperelasticity
+//
+//    Model follows from W = lambda / 2 * (ln J)^2 - mu * ln J + 1/2 * mu *
+//    (tr(C)- I)
+//
+//    with C = 1/2*(F.T*F-I) and J = det(F)
+//
+class NeoHookeanStressUpdate : public ComputeLagrangianStressPK2
+{
+public:
+  static InputParameters validParams();
+  NeoHookeanStressUpdate(const InputParameters & parameters);
+
+protected:
+  /// Actual stress/Jacobian update
+  virtual void computeQpPK2Stress();
+
+protected:
+  const MaterialProperty<Real> & _lambda;
+  const MaterialProperty<Real> & _mu;
```

Comment:
Instead of two `Real` properties, can we provide a elasticity tensor here like what you did in `StVenantKirchhoffStressUpdate`? Then it will have the flexibility to take any pair of constants (with that Gurantee consumer). You can use `ElasticityTensorTools::getIsotropicShearModulus` to get the shear modulus, and maybe add a `ElasticityTensorTools::getIsotropicLameParameter` to get lambda.

Reply:
This one really doesn't work like `StVenantKirchoffStressUpdate` -- there's no real association with the standard linear elasticity except for the linearized form.  My implementation with the shear and bulk modulii is actually pretty unusual, normally you just use the "C and D" constants.  We could do it the way you suggest, but it's probably not going to be too helpful for users, unlike `StVenantKirchooffStressUpdate`.

Reply:
You mean the linearized form of this Neo-Hookean model doesn't have the usual conversion among lame parameters? I didn't know that. But if that's case, then I'm fine with what's being done here.